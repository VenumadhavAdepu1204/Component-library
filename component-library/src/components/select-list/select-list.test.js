import React from 'react'
import '@testing-library/react/cleanup-after-each'
import '@testing-library/jest-dom/extend-expect'
import { render, fireEvent } from '@testing-library/react'
import SelectList from './index'

describe('Select List', () => {
  test('Select List renders properly without any error', () => {
    const testLabel = [{ label: 'label1' }, { label: 'label2' }]
    const { container } = render(<SelectList navListData={testLabel} />)
    expect(container).toMatchSnapshot()
  })
  test('Select List renders properly with empty data', () => {
    const testLabel = []
    const { container } = render(<SelectList navListData={testLabel} />)
    expect(container).toMatchSnapshot()
  })
  test('Select List renders properly when props not passed', () => {
    const { container } = render(<SelectList />)
    expect(container).toMatchSnapshot()
  })
  test('Select List invoked with click', () => {
    const testLabel = [{ label: 'label1' }]
    const { getByTestId } = render(<SelectList navListData={testLabel} />)
    const listDiv = getByTestId('option_label1')
    fireEvent.click(listDiv)
  })
})
