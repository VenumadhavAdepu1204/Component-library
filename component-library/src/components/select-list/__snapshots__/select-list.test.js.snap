// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select List Select List renders properly when props not passed 1`] = `<div />`;

exports[`Select List Select List renders properly with empty data 1`] = `<div />`;

exports[`Select List Select List renders properly without any error 1`] = `
<div>
  <a
    aria-label="linklabel1"
    class="sc-bwzfXH kyHOWt"
    data-test-id="option_label1"
    data-testid="option_label1"
    href="/#/label1"
    id="option_label1"
    tabindex="0"
  >
    label1
    <span
      class="sc-bdVaJa cFwaQV"
    >
      <svg
        data-test-id="undefined-icon-svg"
        focusable="false"
        height="1em"
        viewBox="0 0 32 32"
        width="1em"
      >
        <path
          d="M10.675,25.4380505 C9.775,24.6380505 9.775,23.2380505 10.675,22.3380505 L16.975,16.0380505 L10.675,9.73805052 C9.775,8.93805052 9.775,7.53805052 10.675,6.63805052 C11.575,5.73805052 12.875,5.83805052 13.775,6.63805052 L21.575,14.4380505 C22.475,15.2380505 22.475,16.6380505 21.575,17.5380505 L13.775,25.3380505 C12.975,26.3380505 11.575,26.3380505 10.675,25.4380505 Z"
          data-test-id="undefined-icon-path"
          fill="#0093D8"
          id="undefined-data-id"
        />
      </svg>
    </span>
  </a>
  <a
    aria-label="linklabel2"
    class="sc-bwzfXH kyHOWt"
    data-test-id="option_label2"
    data-testid="option_label2"
    href="/#/label2"
    id="option_label2"
    tabindex="0"
  >
    label2
    <span
      class="sc-bdVaJa cFwaQV"
    >
      <svg
        data-test-id="undefined-icon-svg"
        focusable="false"
        height="1em"
        viewBox="0 0 32 32"
        width="1em"
      >
        <path
          d="M10.675,25.4380505 C9.775,24.6380505 9.775,23.2380505 10.675,22.3380505 L16.975,16.0380505 L10.675,9.73805052 C9.775,8.93805052 9.775,7.53805052 10.675,6.63805052 C11.575,5.73805052 12.875,5.83805052 13.775,6.63805052 L21.575,14.4380505 C22.475,15.2380505 22.475,16.6380505 21.575,17.5380505 L13.775,25.3380505 C12.975,26.3380505 11.575,26.3380505 10.675,25.4380505 Z"
          data-test-id="undefined-icon-path"
          fill="#0093D8"
          id="undefined-data-id"
        />
      </svg>
    </span>
  </a>
</div>
`;
